lib/hashgraph.cc

```diff
@@ -276,29 +276,22 @@ void Hashgraph::consume_sequence_and_tag(const std::string& seq,
 //
 
 // TODO? Inline in header.
-void
-Hashgraph::
-consume_fasta_and_tag(
-    std:: string const  &filename,
-    unsigned int	      &total_reads, unsigned long long	&n_consumed
+template<typename SeqIO>
```

Comment:
My ultimate plan is to change `consume_fasta` and company to `consume_seq` or `consume_seqfile` or `consume_infile` or something like that (suggestions welcome), with ReadParser handling the generic stuff and templated readers handling the format-specific sequence loading.
When all is said and done the function instantiations would look like `Hashtable::consume_seqfile<FastxReader>` and `Hashtable::consume_seqfile<BamReader>`.
However, I thought it better to split this up into three PRs rather than a single one.

- In the first PR, just simply replace the existing functionality with the new data structure and template framework.
- In the second PR, replace the `consume_fasta` function names with something more generic.
- In the third PR, introduce the BAM reader and extend the Python API to enable parsing Fastx or BAM.

Yes, this all could be done with inheritance, but I figured object composition and templating would enable better performance (handle polymorphism at compile time rather than runtime).